block/manager.go

```diff
@@ -351,26 +351,27 @@ func (m *Manager) sendNonBlockingSignalToRetrieveCh() {
 	}
 }
 
-// trySyncNextBlock tries to progress one step (one block) in sync process.
+// trySyncNextBlock tries to execute as many blocks as possible.
 //
-// To be able to apply block and height h, we need to have its Commit. It is contained in block at height h+1.
-// If block at height h+1 is not available, value of last gossiped commit is checked.
+// For every block, to be able to apply block at height h, we need to have its Commit. It is contained in block at height h+1.
 // If commit for block h is available, we proceed with sync process, and remove synced block from sync cache.
 func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
-	var commit *types.Commit
-	currentHeight := m.store.Height() // TODO(tzdybal): maybe store a copy in memory
-
-	b, ok := m.blockCache.getBlock(currentHeight + 1)
-	if !ok {
-		return nil
-	}
+	for {
+		currentHeight := m.store.Height() // TODO(tzdybal): maybe store a copy in memory
+		b, ok := m.blockCache.getBlock(currentHeight + 1)
+		if b == nil || !ok {
+			return nil
+		}
 
-	signedHeader := &b.SignedHeader
-	if signedHeader != nil {
-		commit = &b.SignedHeader.Commit
-	}
+		var commit *types.Commit
+		signedHeader := &b.SignedHeader
+		if signedHeader != nil {
+			commit = &b.SignedHeader.Commit
+		}
 
-	if b != nil && commit != nil {
+		if commit == nil {
+			return nil
+		}
```

Comment:
We should have a comment here as to why we would return instead of continue now that this is an infinite loop.

Also we only use `commit` in one place, so defining a separate variable seems unnecessary. I would just check `b.SignedHeader == nil` directly for the return/continue case.